gensim/models/wrappers/ldamallet.py

```diff
@@ -245,11 +248,40 @@ def show_topic(self, topicid, topn=10):
     def print_topic(self, topicid, topn=10):
         return ' + '.join(['%.3f*%s' % v for v in self.show_topic(topicid, topn)])
 
+
+    def get_version(self, direc_path):
+        """"
+
+        function to return the version of `mallet`
+
+        """
+        try:
+            """
+            Check version of mallet via jar file
+            """
+            archive = zipfile.ZipFile(direc_path, 'r')
```

Comment:
`smart_open` doesn't handle zip files (=multiple files in a single archive). It only handles file compression, not multi-file archives.